### PR TITLE
Add HyperFileConfig with options for timestamptz, allow_nulls, and decimal conversion

### PR DESCRIPTION
**Pull request by dustinvannoy-db**
These changes are to provide a few options to work around issues with timestamps and null integers.

1. Add timestamp_with_timezone option to HyperFile and Creator. Defaults to False to keep backward compatibility, but setting this to `True` avoids error:
    ```
     HyperException: Type 'timestamp' is incompatible to type 'timestamptz' of column 'last_updated' in Parquet file.
    ``` 
2. Add allow_nulls to skip clean_dataframe step which replaces empty/null numbers with 0.

### CODE REVIEW

**Comment by dustinvannoy-db** [comment]
@goodwillpunning I will schedule time to talk through this. Still need to build and have customer test before merging but wanted to get your eyes on it.

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+[build-system]
+requires = ["hatchling"]
+build-backend = "hatchling.build"
+
+[project]
+name = "hyperleaup"
+version = "0.1.0"
```
Comment (goodwillpunning) [inline pyproject.toml]: @dustinvannoy-db could you please bump the version up to `0.1.1` ?

**Review by goodwillpunning — APPROVED** [review]
LGTM! 👍 Thanks for your code contribution!

**Review by goodwillpunning — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**